src/test/kotlin/com/hibob/academy/service/UserServiceTest.kt

```diff
@@ -0,0 +1,87 @@
+package com.hibob.academy.service
+
+import com.hibob.bootcamp.unittests.*
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+class UserServiceTest {
+    private val userDao = mock<UserDao> {}
+    private val notificationService = mock<NotificationService>()
+    private val emailVerificationService = mock<EmailVerificationService>()
+    private val userService = UserService(userDao, notificationService, emailVerificationService)
+
+    @Test
+    fun `try to insert user who already exsits`() {
+        val user = User(1, "bob", "bob@hibob.com", "123456", true)
+        whenever(userDao.findById(1)).thenReturn(user)
+        assertThrows<IllegalArgumentException> { userService.registerUser(user) }
```

Comment:
assertThrows returns the exception, so you can also test that the exception message is correct